oci/pull.bzl

```diff
@@ -37,24 +37,164 @@ oci_image(
 """
 
 load("@aspect_bazel_lib//lib:paths.bzl", "BASH_RLOCATION_FUNCTION")
+load("@aspect_bazel_lib//lib:base64.bzl", "base64")
+load("@aspect_bazel_lib//lib:repo_utils.bzl", "repo_utils")
+
+def _strip_host(url):
+    # TODO: a principled way of doing this
+    return url.replace("http://", "").replace("https://", "").replace("/v1/", "")
+
+def _file_exists(rctx, path):
+    result = rctx.execute(["stat", path])
+    return result.return_code == 0
+
+def _get_auth_file_path(rctx):
+    # this is the standard path where registry credentials are stored
+    config_path = "{}/.docker/config.json".format(rctx.os.environ["HOME"])
+
+    # set config path to DOCKER_CONFIG env if present
+    if "DOCKER_CONFIG" in rctx.os.environ:
+        config_path = rctx.os.environ["DOCKER_CONFIG"]
+
+    if _file_exists(rctx, config_path):
+        return config_path
+
+    # https://docs.podman.io/en/latest/markdown/podman-login.1.html#authfile-path
+    XDG_RUNTIME_DIR = "{}/.config".format(rctx.os.environ["HOME"])
+    if "XDG_RUNTIME_DIR" in rctx.os.environ:
+        XDG_RUNTIME_DIR = rctx.os.environ["XDG_RUNTIME_DIR"]
+
+    config_path = "{}/containers/auth.json".format(XDG_RUNTIME_DIR)
+
+    if _file_exists(rctx, config_path):
+        return config_path
+
+    return None
+
+def _auth_anonymous(rctx, registry, repository, identifier):
```

Comment:
this is specific to dockerhub right? I know the function is basically no-op when the registry isn't `index.docker.io` but to me that suggests that the caller should be the one checking the registry argument and not call this function.

(unless you think we'll add detection for other registries in here)

Reply:
Yes. we'll probably add other registries here as people come up with new ones. Unfortunately, we can't generalize this because we can't read the WWW-Authenticate header.